sites/org/hooks/use-navigation.mjs

```diff
@@ -169,43 +171,50 @@ const sitePages = (t = false, control = 99) => {
   return pages
 }
 
-export const useNavigation = (param = {}, extra = []) => {
-  const { ignoreControl = false } = param
+export const useNavigation = ({ ignoreControl = false }) => {
   // Passing in the locale is not very DRY so let's just grab it from the router
   const { locale } = useRouter()
   // We need translation
   const { t } = useTranslation(ns)
   // We need the account if we want to take control into account
   const { account } = useAccount()
 
-  const siteNav = {
-    ...pbn[locale],
-    ...sitePages(t, ignoreControl ? undefined : account.control),
-  }
-  for (const [_path, _data] of extra) {
-    objUpdate(siteNav, _path, _data)
-  }
+  const [navigation, setNavigation] = useState({
```

Comment:
We could move it to its own context, but that felt wasteful to me. What I found was that we were calling `useNavigation` in lots of places where we wanted the same (massive) structure that almost never changes, and it's not cached so it was being calculated every time. You definitely helped with that when you refactored `ReadMore`, but there were still plenty of places where it made sense to me to say "we already made that data, let me access the existing one". In most cases, the `NavigationContext` was already also being used, usually to get, rather than set navigation state, and it was already calling `useNavigation` internally